src/traces/bar/plot.js

```diff
@@ -226,6 +226,21 @@ function appendBarText(gd, bar, calcTrace, i, x0, x1, y0, y1) {
     var insideTextFont = style.getInsideTextFont(trace, i, layoutFont, barColor);
     var outsideTextFont = style.getOutsideTextFont(trace, i, layoutFont);
 
+    // Special case: don't use the c2p(v, true) value on log size axes,
```

Comment:
See 

https://github.com/plotly/plotly.js/blob/1d0acf035ac054c983c3e6bb1b4a691a0c8cc8be/src/plots/cartesian/set_convert.js#L68-L80

for more info that `true` option.